apps/web/utils/middleware.ts

```
@@ -14,12 +14,15 @@ import { env } from "@/env";
 
-export type NextHandler = (
+type NextHandler<TContext extends Record<string, any>, TResponse = void> = (
   req: NextRequest,
-  { params }: { params: Record<string, string | undefined> },
-) => Promise<NextResponse | StreamingTextResponse>;
+  context: { params: Promise<TContext> },
+) => Promise<NextResponse<TResponse> | StreamingTextResponse>;
 
-export function withError(handler: NextHandler): NextHandler {
+export function withError<
+  TContext extends Record<string, any>,
```

Biome

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)